cpp/src/arrow/compute/api_vector.cc

```diff
@@ -140,6 +150,50 @@ PartitionNthOptions::PartitionNthOptions(int64_t pivot)
     : FunctionOptions(internal::kPartitionNthOptionsType), pivot(pivot) {}
 constexpr char PartitionNthOptions::kTypeName[];
 
+SelectKOptions::SelectKOptions(int64_t k, std::vector<SortKey> sort_keys,
+                               SelectKAlgorithm kind)
+    : FunctionOptions(internal::kSelectKOptionsType),
+      k(k),
+      sort_keys(std::move(sort_keys)),
+      kind(kind) {}
+
+bool SelectKOptions::is_top_k() const {
+  SortOrder order = SortOrder::Descending;
+  for (const auto& k : sort_keys) {
+    order = k.order;
+    if (order != SortOrder::Descending) {
+      break;
+    }
+  }
+  return order == SortOrder::Descending;
+}
```

Comment:
Why track `order`, instead of just iterating through the sort keys and returning `false` if the order is wrong, returning `true` outside the loop?

Reply:
```suggestion
bool SelectKOptions::is_top_k() const {
  for (const auto& k : sort_keys) {
    if (k.order != SortOrder::Descending) {
      return false;
    }
  }
  return true;
}
```